chain/chain/src/runtime/mod.rs

```diff
@@ -828,11 +828,31 @@ impl RuntimeAdapter for NightshadeRuntime {
     }
 
     fn get_gc_stop_height(&self, block_hash: &CryptoHash) -> BlockHeight {
+        use std::sync::atomic::{AtomicI64, Ordering};
+        use std::time::{SystemTime, UNIX_EPOCH};
+
+        // Use AtomicI64 to store unix timestamp in seconds
+        static LAST_LOG_TIME: AtomicI64 = AtomicI64::new(0);
+        // Log throttling interval (10 seconds)
+        const LOG_THROTTLE_INTERVAL: i64 = 10;
```

Comment:
I would prefer for the values to be `u64` unless we need support for signed ints.